src/ui/data/line-chart.utils.ts

```diff
@@ -0,0 +1,100 @@
+import deepmerge from 'deepmerge'
+import { EChartsOption } from 'echarts'
+import { defaultLineChartOption } from './line-chart.defaults'
+
+export const mergeReplaceArrays = <T>(target: T, ...sources: Partial<T>[]): T =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  deepmerge.all([target as any, ...sources] as any, {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    arrayMerge: (_destinationArray: any[], sourceArray: any[]) => sourceArray,
+  }) as T
```

Comment:
I think in TS there's Array type, maybe you can use this instead of any[]

Reply:
btw. if you're only using this in buildLineChartOption, then maybe move it there & use echarts types